aiidalab/utils.py

```diff
@@ -29,45 +30,30 @@
 logger = logging.getLogger(__name__)
 FIND_INSTALLED_PACKAGES_CACHE = TTLCache(maxsize=32, ttl=60)  # type: ignore
 
-# Warning: try-except is a fix for Quantum Mobile release v19.03.0 where
-# requests_cache is not installed.
-try:
-    import requests_cache
-except ImportError:
-    logger.warning(
-        "The requests_cache package is missing.  Requests made to the app "
-        "registry will not be cached!"
-    )
-else:
-    # Install global cache for all requests.
-
-    # The cache file is placed within the user home directory.
-    cache_file = Path.home().joinpath(".cache", "requests", "cache")
-    cache_file.parent.mkdir(parents=True, exist_ok=True)
-
-    # The cache is configured to avoid spamming the index server with requests
-    # that are made in rapid succession and also serves as a fallback in case
-    # that the index server is not reachable.
-    requests_cache.install_cache(
-        cache_name=str(cache_file),
-        backend="sqlite",
-        expire_after=60,  # seconds
-        old_data_on_error=True,
-    )
+# The cache is configured to avoid spamming the app registry server with requests
+# that are made in rapid succession and also serves as a fallback in case
+# that the index server is temporarily not reachable.
+_session = CachedSession(
```

Comment:
Per  https://requests-cache.readthedocs.io/en/stable/user_guide/general.html#sessions

The old global approach is described here:
https://requests-cache.readthedocs.io/en/stable/user_guide/general.html#patching

and it's downsides, which specifically mention multithreaded use: https://requests-cache.readthedocs.io/en/stable/user_guide/general.html#patching-limitations-potential-issues